nativescript-angular/directives/dialogs.ts

```diff
@@ -17,15 +18,15 @@ import { AppHostView } from "../app-host-view";
 import { DetachedLoader } from "../common/detached-loader";
 import { PageFactory, PAGE_FACTORY } from "../platform-providers";
 import { once } from "../common/utils";
-import { topmost, Frame } from "tns-core-modules/ui/frame";
+import { topmost, Frame, ShowModalOptions } from "tns-core-modules/ui/frame";
 
-export interface ModalDialogOptions {
+export type BaseShowModalOptions = Pick<ShowModalOptions, Exclude<keyof ShowModalOptions, "closeCallback" | "context">>;
+
+export interface ModalDialogOptions extends BaseShowModalOptions {
     context?: any;
-    fullscreen?: boolean;
-    animated?: boolean;
-    stretched?: boolean;
     viewContainerRef?: ViewContainerRef;
     moduleRef?: NgModuleRef<any>;
+    sourceView?: ElementRef;
```

Comment:
A better name for this would be `targetView` or `target` and we have discussed internally that it would be better if it's of type `View`. There are different ways to get to the NS View in Angular and we wouldn't want the developer to have to get to an `ElementRef` to use this API.

Reply:
I like the idea of it being a of `View` type. This was made in the previous PR from where I based this PR. I think that simple the name `target` fits best as this object is going to be of type `View` there is not need to include that in its signature.